core/runtime/src/main/java/io/quarkus/runtime/ApplicationConfig.java

```diff
@@ -22,4 +22,11 @@
      */
     @ConfigItem
     public Optional<String> version;
+
+    /**
+     * The header to use for UI Screen (Swagger UI, GraphQL UI etc)
+     * If not set, defaults to {applicationName} {applicationVersion} (powered by Quarkus {quarkusVersion}).
```

Comment:
Why don't you inject the default here instead and let the automatic replacement you already have in place do its thing? This way, it would be automatically documented.

Reply:
Mostly because of the "vars" used. (eg. {applicationName}. But maybe it can be done. Let me have a look. Hold on :)

Reply:
Ok yea, this is actually better, as now the user can also use these "vars" in their config if they want.